Model/Media.php

```diff
@@ -646,7 +646,7 @@ public function getCategory()
     public function setCategory($category = null)
     {
         if (null !== $category && $category instanceof CategoryInterface) {
-            throw new UnexpectedTypeException();
+            throw new \InvalidArgumentException('Argument 1 should be an instance of Sonata\ClassificationBundle\Model\CategoryInterface or null');
```

Comment:
I put the line like the exception trow some lines before (line 622). I can change it but this will be seen different.